src/averageAges.js

```diff
@@ -51,11 +63,33 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @param {object[]} people
  * @param {boolean} onlyWithSon - optional
- *
+ *~
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => !onlyWithSon
+    // here we find mother and her child
+    ? child.mother && people.find(person => person.name === child.mother)
+    // here we find mother only with son
+    : child.mother && people.find(person => person.name === child.mother)
+     && child.sex === 'm');
+
+  const ageDiff = children.reduce((x, child) => {
+    // here we find all mothers
+    const mother = people.find(person => person.name === child.mother);
+
+    // here we find age difference between mothers & childrens
+    return x + child.born - mother.born;
+  }, 0);
+
+  return ageDiff / children.length;
+}
+
+function findAvg(people) {
+  // this is our function for finding an average of people age.
+  const avgAge = people.reduce((x, year) => x + (year.died - year.born), 0);
```

Comment:
here i think 'sumOfAge' will be better because you count total age here

Reply:
alright